// dexguru-sdk/dist/models/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.api_public__models_api__rest_token__RestTokenFinanceModel__SortFields = exports.api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields = exports.TransactionChoices = exports.TokenTradeDirections = exports.OrderChoices = exports.ChainChoices = exports.CategoriesChoices = exports.AmmChoices = void 0;
var AmmChoices;
(function (AmmChoices) {
    AmmChoices["all"] = "all";
    AmmChoices["uniswap_v3"] = "uniswap_v3";
    AmmChoices["uniswap"] = "uniswap";
    AmmChoices["uniswap_v2"] = "uniswap_v2";
    AmmChoices["pancakeswap"] = "pancakeswap";
    AmmChoices["sushiswap"] = "sushiswap";
    AmmChoices["quickswap"] = "quickswap";
    AmmChoices["pangolin"] = "pangolin";
    AmmChoices["traderjoe"] = "traderjoe";
    AmmChoices["ubeswap"] = "ubeswap";
    AmmChoices["spookyswap"] = "spookyswap";
    AmmChoices["spiritswap"] = "spiritswap";
    AmmChoices["kyber"] = "kyber";
    AmmChoices["lydia"] = "lydia";
})(AmmChoices = exports.AmmChoices || (exports.AmmChoices = {}));
var CategoriesChoices;
(function (CategoriesChoices) {
    CategoriesChoices["noob"] = "noob";
    CategoriesChoices["casual"] = "casual";
    CategoriesChoices["medium"] = "medium";
    CategoriesChoices["heavy"] = "heavy";
    CategoriesChoices["bot"] = "bot";
})(CategoriesChoices = exports.CategoriesChoices || (exports.CategoriesChoices = {}));
var ChainChoices;
(function (ChainChoices) {
    ChainChoices["_1"] = "1";
    ChainChoices["_56"] = "56";
    ChainChoices["_137"] = "137";
    ChainChoices["_43114"] = "43114";
    ChainChoices["_42161"] = "42161";
    ChainChoices["_42220"] = "42220";
    ChainChoices["_250"] = "250";
})(ChainChoices = exports.ChainChoices || (exports.ChainChoices = {}));
var OrderChoices;
(function (OrderChoices) {
    OrderChoices["asc"] = "asc";
    OrderChoices["desc"] = "desc";
})(OrderChoices = exports.OrderChoices || (exports.OrderChoices = {}));
var TokenTradeDirections;
(function (TokenTradeDirections) {
    TokenTradeDirections["in"] = "in";
    TokenTradeDirections["out"] = "out";
})(TokenTradeDirections = exports.TokenTradeDirections || (exports.TokenTradeDirections = {}));
var TransactionChoices;
(function (TransactionChoices) {
    TransactionChoices["swap"] = "swap";
    TransactionChoices["burn"] = "burn";
    TransactionChoices["mint"] = "mint";
})(TransactionChoices = exports.TransactionChoices || (exports.TransactionChoices = {}));
var api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields;
(function (api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields) {
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["transaction_address"] = "transaction_address";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["timestamp"] = "timestamp";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["block_number"] = "block_number";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["to"] = "to";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["sender"] = "sender";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["amount_usd"] = "amount_usd";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["pair_address"] = "pair_address";
    api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields["wallet_address"] = "wallet_address";
})(api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields = exports.api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields || (exports.api_public__models_api__rest_swaps_burns_mints__RestSwapBurnMintModel__SortFields = {}));
var api_public__models_api__rest_token__RestTokenFinanceModel__SortFields;
(function (api_public__models_api__rest_token__RestTokenFinanceModel__SortFields) {
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["address"] = "address";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["volume_24h"] = "volume_24h";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["liquidity"] = "liquidity";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["volume_24h_usd"] = "volume_24h_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["liquidity_usd"] = "liquidity_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["price_usd"] = "price_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["volume_24h_delta"] = "volume_24h_delta";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["liquidity_24h_delta"] = "liquidity_24h_delta";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["price_24h_delta"] = "price_24h_delta";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["volume_24h_delta_usd"] = "volume_24h_delta_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["liquidity_24h_delta_usd"] = "liquidity_24h_delta_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["price_24h_delta_usd"] = "price_24h_delta_usd";
    api_public__models_api__rest_token__RestTokenFinanceModel__SortFields["timestamp"] = "timestamp";
})(api_public__models_api__rest_token__RestTokenFinanceModel__SortFields = exports.api_public__models_api__rest_token__RestTokenFinanceModel__SortFields || (exports.api_public__models_api__rest_token__RestTokenFinanceModel__SortFields = {}));
